dmutils/email/dm_mailchimp.py

```diff
@@ -93,17 +94,20 @@ def subscribe_new_email_to_list(self, list_id, email_address):
             # mailchimp to never add them to mailchimp lists. In this case, we resort to allowing a failed API call (but
             # log) as a user of this method would unlikely be able to do anything as we have no control over this
             # behaviour.
-            if getattr(e, "response", None) is not None and (
-                "looks fake or invalid, please enter a real email address." in e.response.json().get("detail", "")
-            ):
-                self.logger.error(
-                    "Expected error: Mailchimp failed to add user ({}) to list ({}). API error: The email address looks fake or invalid, please enter a real email address.".format(  # noqa
-                        hashed_email,
-                        list_id
-                    ),
-                    extra={"error": str(e)}
-                )
-                return True
+            try:
+                if getattr(e, "response", None) is not None and (
+                    "looks fake or invalid, please enter a real email address." in e.response.json().get("detail", "")
+                ):
+                    self.logger.error(
+                        "Expected error: Mailchimp failed to add user ({}) to list ({}). API error: The email address looks fake or invalid, please enter a real email address.".format(  # noqa
+                            hashed_email,
+                            list_id
+                        ),
+                        extra={"error": str(e)}
+                    )
+                    return True
```

Comment:
I'm confused by what this function returns. Why True in this case?

Reply:
I'll have a look, but is the logic not 'Return False in all failure cases except this one special reason, then return True'? Which is what it still does?

Reply:
Yeah ok I think I can see how that can make sense now :-)

Reply:
Cool. I'll still have a dig though. Would be good to know.